seatunnel-api/src/main/java/org/apache/seatunnel/api/table/type/SeaTunnelRow.java

```diff
@@ -97,6 +100,180 @@ public boolean isNullAt(int pos) {
         return this.fields[pos] == null;
     }
 
+    public int getBytesSize(SeaTunnelRowType rowType) {
+        if (size == 0) {
+            int s = 0;
+            for (int i = 0; i < fields.length; i++) {
+                s += getBytesForValue(fields[i], rowType.getFieldType(i));
+            }
+            size = s;
+        }
+        return size;
+    }
+
+    /** faster version of {@link #getBytesSize(SeaTunnelRowType)}. */
+    private int getBytesForValue(Object v, SeaTunnelDataType<?> dataType) {
+        if (v == null) {
+            return 0;
+        }
+        SqlType sqlType = dataType.getSqlType();
+        switch (sqlType) {
+            case STRING:
+                return ((String) v).length();
```

Comment:
`length` does not return byte length, it will return the unique code length.
But it seems we don't  have a better way to get the byte length.